src/main/java/org/openpnp/vision/pipeline/CvPipeline.java

```diff
@@ -238,25 +295,43 @@ public void process() {
 
             // If the result image is null and there is a working image, replace the result image
             // replace the result image with a clone of the working image.
-            if (image == null) {
-                if (workingImage != null) {
-                    image = workingImage.clone();
-                }
-            }
-            // If the result image is not null:
-            // Release the working image if the result image is different.
-            // Replace the working image with the result image.
-            // Clone the result image for storage.
-            else {
-                if (workingImage != null && workingImage != image) {
-                    workingImage.release();
-                }
+            if (image != null && workingImage!=image ) {
                 workingImage = image;
-                image = image.clone();
+				if(!workingImages.contains(image)) {
+					workingImages.add(image);
+				}
             }
-
-            results.put(stage, new Result(image, model, processingTimeNs));
+			String id = stage.getName();
+			result= new Result(workingImage, model, processingTimeNs = System.nanoTime() - processingTimeNs);
+			if(id.startsWith("result")&&id.length()==7) {
```

Comment:
This is sloppy. Hardcoding a format for stage names and having special functionality does not improve the system, it makes it more fragile and increases the code complexity. If you want to be able to handle multiple result stages it needs to be thought through better, and should probably just be handled in the calling code.